src/shared/Performance.js

```diff
@@ -0,0 +1,20 @@
+export class Performance {
+  static hasPerformanceApi () {
+    const p = window.performance
+    return p && p.mark && p.measure && p.clearMarks && p.clearMeasures
+  }
+
+  static mark (tag) {
+    if (Performance.hasPerformanceApi()) {
```

Comment:
why `Performance` and not `this`?

Reply:
As the methods are all `static` they must be called statically. I just wrapped them into a class to make them more readable at their usage points.